packages/framework-core/src/decorators/metadata.ts

```diff
@@ -10,3 +10,33 @@ export function getPropertiesMetadata(classType: AnyClass): Array<PropertyMetada
   }
   return meta.fields
 }
+
+/**
+ * Get the argument names from a given function.
+ *
+ * This implementation is a TypeScript adaptation of a JavaScript implementation
+ * borrowed from the promisify-node code and can be found in the following link:
+ * https://github.com/nodegit/promisify-node/blob/02fc47cfc00146a533193bc4740e2e3e3be81c6f/utils/args.js
+ *
+ * @param {Function} func - The function to parse.
+ * @returns {Array} arg - List of arguments in the function.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function getFunctionArguments(func: Function): Array<string> {
+  // First match everything inside the function argument parens.
+  const args = func.toString().match(/([^(])*\(([^)]*)\)/)?.[2]
```

Comment:
I would replace the regex code for a parser with [`parser-ts`](https://github.com/gcanti/parser-ts) but we can leave this for a future iteration